ocw_oer_export/client.py

```diff
@@ -0,0 +1,37 @@
+"""
+Module for interacting with the MIT OpenCourseWare API.
+"""
+import logging
+import requests
+from retry import retry
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+@retry(tries=3, delay=2, logger=logger)
+def fetch_data_from_api(api_url, limit=100):
```

Comment:
**Issue?:** I don't think this retry is working as expected. It's supposed to retry API calls if they fail (e.g., due to intermittent network issues), right? But the API call is made in a try/request, so I don't think the retries ever get triggered, since there's no error? And I think that we probably want the retry on individual API calls, not on the whole loop.



Separately: A nice approach might be to make "paginator" like django has (https://docs.djangoproject.com/en/5.0/topics/pagination/#the-paginator-class). The idea would be something like
```python
pages = paginated_response(page_size)
for page in pages:
   # do whatever with the page
```
where each page is an API call.

Something like
```python
def paginated_response(api_url, page_size=100):

    next = api_url
    while next:
        response = make_request(next, page_size)    # <------ retry on `make_request` 
        data = response.json()
        results = data.get("results", [])
        next = data.get("next")
        yield results
```